dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBMongoDB/AzureCosmosDBMongoDBMemoryStore.cs

```diff
@@ -48,15 +48,13 @@ AzureCosmosDBMongoDBConfig config
     /// and other properties required for vector search.
     /// </summary>
     public AzureCosmosDBMongoDBMemoryStore(
-        IMongoClient mongoClient,
+        MongoClient mongoClient,
         string databaseName,
         AzureCosmosDBMongoDBConfig config
     )
     {
-        MongoClientSettings settings = mongoClient.Settings;
         this._config = config;
-        settings.ApplicationName = this._config.ApplicationName;
-        this._mongoClient = new MongoClient(settings);
+        this._mongoClient = mongoClient;
```

Comment:
Previous logic worked because even though user passes `mongoClient`, we created our own instance of `mongoClient` based on settings. Now, we are just setting it as it is, and this is a problem, because we have a logic that disposes `mongoClient.Cluster`. The rule is - if we didn't create it, we should not dispose it. You can check how it's handled in Kusto or Milvus connectors.

Reply:
Fixed in next iteration.

Reply:
That makes sense, but since with the new code we are using the client that was passed in constructor, we cannot dispose it now in `Dispose` method. The following logic should be added:
1. When user passes `connectionString` in constructor - we create our own client, use it for operations and we dispose it.
2. When user passes instance of `MongoClient` - we use this client for operations, and we do not dispose it.

Reply:
Adding a boolean global flag **_disposing** which is set True when user passes connectionString 

1. Set to True  - When user passes `connectionString` in constructor
2. Set to False - When user passes instance of `MongoClient`

And use this flag in the Dispose method call.

Reply:
Sounds good, thank you. By the way, no need to set it to `false` explicitly, since it's `false` by default. Setting it to `true` when user passes `connectionString` should be enough. Thanks again!